lib/src/actions/user/providers/linux.rs

```diff
@@ -0,0 +1,51 @@
+use super::UserProvider;
+use crate::steps::Step;
+use crate::{actions::user::UserVariant, atoms::command::Exec};
+use serde::{Deserialize, Serialize};
+use which::which;
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub struct LinuxUserProvider {}
+
+impl UserProvider for LinuxUserProvider {
+    fn add_user(&self, user: &UserVariant) -> Vec<Step> {
+        let mut args: Vec<String> = vec![];
+
+        // is a user name isn't provided, cant create a new user
+        if user.username.is_empty() {
+            return vec![];
+        }
+
+        args.push(user.username.clone());
+
+        if !user.home_dir.is_empty() {
+            args.push(String::from("-m"));
+            args.push(String::from("-d"));
+            args.push(user.home_dir.clone());
+        }
+
+        if !user.shell.is_empty() {
+            args.push(String::from("-s"));
+            args.push(user.shell.clone());
+        }
+
+        if !user.fullname.is_empty() {
+            args.push(String::from("-c"));
+            args.push(user.fullname.clone());
+        }
+
+        vec![Step {
+            atom: Box::new(Exec {
+                command: String::from(which::which("useradd").unwrap().to_str().unwrap()),
```

Comment:
I would rather put the `which` call to the top ot the `add_user` method and use a `if let Ok(path) = which::which("useradd")` construct or a `match`. Much safer than `.unwrap`

Reply:
Some Linux systems use useradd and some use adduser ...

Reply:
I'll need to remember which distro uses which for testing. Hopefully the parameters are roughly the same, if not will have to branch execution

Reply:
Swapped over to which statement. 

Looking at useradd vs adduser. All distros should have `useradd` since this is the binary compiled with the distros. `adduser` is a perl script that makes adding a user a little easier and uses `useradd` on the back end. 

https://linuxconfig.org/add-user-linux-command